lib/src/main/kotlin/at/bitfire/synctools/icalendar/validation/StreamPreprocessor.kt

```diff
@@ -6,51 +6,14 @@
 
 package at.bitfire.synctools.icalendar.validation
 
-import java.io.IOException
-import java.io.Reader
-import java.io.StringReader
-import java.util.Scanner
-
-abstract class StreamPreprocessor {
-
-    abstract fun regexpForProblem(): Regex?
+interface StreamPreprocessor {
 
     /**
      * Fixes an iCalendar string.
      *
-     * @param original The complete iCalendar string
-     * @return The complete iCalendar string, but fixed
+     * @param lines The iCalendar lines to fix. Those may be the full iCalendar file, or just a part of it.
```

Comment:
We should explicitly mention that `lines` must only contain complete lines. Because when we have to add another `StreamPreprocessor` at some day, we won't know anymore what we did now, but we still need to know whether our input (`lines`) can contain parts of lines or only complete lines.